app/src/main/java/com/concordia/mcga/utilities/pathfinding/PathFinder.java

```diff
@@ -0,0 +1,103 @@
+     * @param destY - ending position's y coordinate
+     * @return - Returns a list of the tiles found in the shortest path. Sorted from first to last.
+     * @throws Exception - Thrown if there exists no valid path between both points
+     */
+    public List<PathFinderTile> shortestPath(int startX, int startY, int destX, int destY)
+        throws Exception {
+        map.setStartTile(startX, startY);
+        map.setEndTile(destX, destY);
+        openSet.add(map.getStartTile());
+        PathFinderTile current;
+        while (true) {
+            current = lowestOpen();
+            if (current.equals(map.getEndTile())) {
+                break;
+            }
+            nextIteration(current);
+        }
+        // Current is now the destination tile. Path is available by traversing parents.
+        List<PathFinderTile> returnList = new ArrayList<>();
+        while (current != null) {
+            returnList.add(0, current);
+            current = current.getParent();
+        }
+        return returnList;
+    }
+
+    /**
+     * Runs an iteration of the A* algorithm.
+     * @param current - Currently examined tile.
+     * @throws Exception
+     */
+    private void nextIteration(PathFinderTile current) throws Exception {
+        openSet.remove(current);
+        closedSet.add(current);
+
+        PathFinderTile[] adjacentTiles = new PathFinderTile[4];
+        adjacentTiles[0] = map.getTile(current.getCoordinateX() + 1, current.getCoordinateY());
+        adjacentTiles[1] = map.getTile(current.getCoordinateX() - 1, current.getCoordinateY());
+        adjacentTiles[2] = map.getTile(current.getCoordinateX(), current.getCoordinateY() - 1);
+        adjacentTiles[3] = map.getTile(current.getCoordinateX(), current.getCoordinateY() + 1);
+
+        for (PathFinderTile tile : adjacentTiles) {
+            if (tile == null) {
+                continue;
+            }
+            if (!closedSet.contains(tile)) {
+                if (openSet.contains(tile)) {
+                    int newDist = tile.calculateDistFromStart();
+                    if (newDist < tile.getDistFromStart()) {
+                        tile.setDistFromStart(newDist);
+                        tile.setParent(current);
+                        openSet.remove(tile);
+                        openSet.add(tile);
+                    }
+                } else {
+                    tile.setParent(current);
+                    tile.setDistFromStart(tile.calculateDistFromStart());
+                    openSet.add(tile);
+                }
+            }
+        }
+    }
+
+    private PathFinderTile lowestOpen() throws Exception {
+        PathFinderTile lowest = PathFinderTile.MAX_COST;
+        for (PathFinderTile tile : openSet) {
+            if (tile.getCost() < lowest.getCost()) {
+                lowest = tile;
+            }
+        }
+        if (lowest.equals(PathFinderTile.MAX_COST)) {
+            throw new Exception("No Valid Path Found!");
```

Comment:
Don't create an exception, return null instead and handle the null in the method calling lowestOpen()